### PR TITLE
contrib/go-pg/pg.v10 add options, ServiceName

### PR DESCRIPTION
**Pull request by pzvalenta**
Adds options

Implements #951

### CODE REVIEW

File: contrib/go-pg/pg.v10/pg_go.go [hunk 1]
```diff
@@ -17,12 +17,22 @@ import (
...
+	defaults(cfg)
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	h := &queryHook{
+		cfg: cfg,
+	}
 	log.Debug("contrib/go-pg/pg.v10: Wrapping Database")
-	db.AddQueryHook(&queryHook{})
+	db.AddQueryHook(h)
```
Comment (knusbaum) [inline contrib/go-pg/pg.v10/pg_go.go:30]: I don't think we need to declare `h`. 
`db.AddQueryHook(&queryHook{cfg: cfg})` is clear enough.

**Review by knusbaum — CHANGES_REQUESTED** [review]
Hi, @pzvalenta. Thanks for the PR.

It looks mostly good, but I have some style change requests for you.
Also, we should probably add tests for the analytics options. They should be easy to copy from other contrib packages.

File: contrib/go-pg/pg.v10/option.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+)
+
+type clientConfig struct {
+	serviceName   string
+	analyticsRate float64
+}
+
+// ClientOption represents an option that can be used to create or wrap a client.
+type ClientOption func(*clientConfig)
```
[the remark this answers is not in the document]
Comment (pzvalenta) [reply contrib/go-pg/pg.v10/option.go]: fixed, thanks

File: contrib/go-pg/pg.v10/option.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+package pg
+
+import (
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+)
+
+type clientConfig struct {
```
[the remark this answers is not in the document]
Comment (pzvalenta) [reply contrib/go-pg/pg.v10/option.go]: fixed, thanks

File: contrib/go-pg/pg.v10/pg_go_test.go [hunk 4]
```diff
@@ -43,7 +43,7 @@ func TestSelect(t *testing.T) {
 		Database: "postgres",
 	})
 
-	Wrap(conn)
+	Wrap(conn, WithServiceName("go-pg"))
```
[the remark this answers is not in the document]
Comment (pzvalenta) [reply contrib/go-pg/pg.v10/pg_go_test.go]: added test, inspired by gintracing

File: contrib/go-pg/pg.v10/pg_go.go (hunk 1 above)
Comment (pzvalenta) [reply contrib/go-pg/pg.v10/pg_go.go]: fixed, thanks

**Comment by pzvalenta** [comment]
Thanks for your review. Added TestAnalyticsRate test inspired by gorm tests and TestServiceName inspired gintracing tests. Both tests are based on the original TestSelect in this package.

**Pull request merged**